Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2658 -- Add filter by subject to combined search

Resolves [EREGCSC-2658](https://jiraent.cms.gov/browse/EREGCSC-2658)

> [!IMPORTANT]  
> Subjects dropdown must be hidden via `v-if` directive before shipping to `prod`.

**Description**

This PR creates a "Choose Subjects" dropdown list that allows users to filter by subject on the `/search` page.  This component uses a Vuetify `v-menu` component alongside our existing `SubjectsSelector.vue` custom component that is already being used on the `/subjects` page.

This "Choose Subjects" dropdown will be hidden on `prod`; we still need to add counts labels and filter by count.  The back end work for the counts is being completed in [EREGCSC-2323](https://jiraent.cms.gov/browse/EREGCSC-2323).  

Once that backend story has shipped, a final story ([EREGCSC-2875](https://jiraent.cms.gov/browse/EREGCSC-2875)) will be completed to enable count labels and filtering by counts.  2875 will also ensure adequate Cypress and Vitest testing coverage.

**This pull request changes:**

- Splits existing Categories dropdown component into a generic, re-usable dropdown component that can accept specific components related to the content and behavior of the dropdown.
   - `dropdowns/GenericDropdown.vue` encapsulates common `v-select` styles and behaviors common to all instances of `v-select`
   - `dropdowns/Categories.vue` and `dropdowns/CategoriesItem.vue` contain markup, logic, and behavior specific to the behavior and display of the Categories dropdown
- Creates `dropdowns/Subjects.vue` component that uses `v-menu` to create a Subjects-specific dropdown component
- Reuses `SubjectsSelector.vue` and the contents of the subjects dropdown list
- Adds logic, utility methods, and CSS styles where appropriate to implement desired functionality

**Steps to manually verify this change:**

1. Green check marks
2. Subjects dropdown appears on the `/search` page
   - dropdown list is populated with all available subjects
   - subject can be selected from the list and filter the search results
   - clicking the "x" in the subject dropdown activator button removes the subject
   - subjects can be filtered by text string using the input element at the top of the dropdown list
   - All of this can be done using the keyboard (tab, up and down arrows, etc)
3. All side effects outlined in the AC of the [JIRA story](https://jiraent.cms.gov/browse/EREGCSC-2658) work as expected
4. Subjects Selector on the `/subjects` page works as expected without any regressions
